modules/ossm-security-auth-policy.adoc

```diff
@@ -0,0 +1,180 @@
++
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: ingress-policy
+  namespace: istio-system
+spec:
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+  action: DENY
+  rules:
+  - from:
+    - source:
+      ipBlocks: ["1.2.3.4"]
+----
++
+2. Run the following command after you write your resource to create your resource in your namespace. The namespace must match your `metadata.namespace` field.
++
+[source,terminal]
+----
+$ oc create -n istio-system -f <filename> 
+----
++
+3. Consider the following examples for other common configurations.
+
+== Configure intra-project communication
+
+You can use `AuthorizationPolicy` to configure your control plane to allow or deny the traffic communicating with your mesh or services in your mesh. 
+
+=== Restrict access to services outside a namespace
+
+You can deny requests from any source that is not in the `bookinfo` namespace with the following `AuthorizationPolicy` resource example.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+ name: httpbin-deny
+ namespace: bookinfo
+spec:
+ selector:
+   matchLabels:
+     app: httpbin
+     version: v1
+ action: DENY
+ rules:
+ - from:
+   - source:
+       notNamespaces: ["bookinfo"]
+----
+
+=== Creating allow-all and default deny-all authorization policies
+
+The following example shows an allow-all authorization policy that allows full access to all workloads in the `bookinfo` namespace.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-all
+  namespace: bookinfo
+spec:
+  action: ALLOW
+  rules:
+  - {}
+----
+
+The following example shows a policy that denies any access to all workloads in the `bookinfo` namespace.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: deny-all
+  namespace: bookinfo
+spec:
+  {}
+----
+
+== Allow or deny access to the ingress gateway
+
+You can set an authorization policy to add allow or deny lists based on IP addresses.
+
+[source,yaml]
+----
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: ingress-policy
+  namespace: istio-system
+spec:
+  selector:
+    matchLabels:
+      app: istio-ingressgateway
+  action: ALLOW
+  rules:
+  - from:
+    - source:
+       ipBlocks: ["1.2.3.4", "5.6.7.0/24"]
+----
+
+== Set restricted access with JSON Web Token
+
+To configure what can access services in your mesh, you can restrict access with a JSON Web Token (JWT). After authentication, a user or service can access routes, services that are associated with that token. 
+
+1. Create a `RequestAuthentication` resource, which defines the authentication methods that are supported by a workload. The following example accepts a JWT issued by `http://localhost:8080/auth/realms/master`.
```

Comment:
Removed hard coded procedure. 

However, I don't think splitting this into a separate module makes sense here. We're still talking about how to configure role based access control here. If we create a separate module, that sounds like creating a new header based on the underlying technology instead of the user story. To restrict access with JWT, you need two resources.

Reply:
> To restrict access with JWT, you need two resources.

And both of those resources are described under the heading "Restrict access with JSON web token." Which is why I suggested splitting this out into a separate module.  The mod docs task template is "Doing one procedure" and this is the second procedure in this module.  I don't think our user stories are defined enough (or small enough) that we've got a one user story = one module workflow yet.